specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/ManagedInstances.json

```diff
@@ -1248,6 +1248,16 @@
             "name": "ExternalGovernanceStatus",
             "modelAsString": true
           }
+        },
+        "isFreemium": {
+          "description": "Weather or not Managed Instance is freemium.",
+          "type": "boolean"
```

Comment:
>boolean [](http://example.com/codeflow?start=19&length=7)

recommend using "enum" instead of "boolean":
[https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md)

Reply:
I will keep this property as boolean, I don't see a need to add another possible value to this property.

Reply:
If this is a new property, then I would also strongly encourage you to make this a "tier" (or similarly named) property with values like "freemium" as we have had Azure service teams introduce more tiers of their service in the future and changing this from a Boolean at a later time is a breaking change to customers which causes customer dissatisfaction.

Reply:
I understand the advantages of having an enum value instead of a boolean, but I still think that a boolean value is better for this case. 
Freemium is just a flag that we use to determine if we should bill the instance or not. The instance being freemium doesn't prevent it from also being another "tier". 

For example, we have a zone redundant flag at the moment:
![image](https://github.com/Azure/azure-rest-api-specs/assets/135128634/ec1b0a46-9246-4513-af15-8dbe7cf0de37)
If hypothetically this wasn't a boolean but an enum that could have values "Regular" and "ZoneRedundant", if we would add freemium to this enum we would have to add "Freemium" and "ZoneRedundantFreemium" values to this enum. Instance being freemium doesn't always prevent it from being zone redundant.

In this example if someone would add a new tier for MI, they would have to know if that tier would ever be allowed for freemium instances and add new values to the enum according to that, which is not ideal.